teleport_16: 16.5.13 -> 16.5.15; teleport_17: 17.5.4 -> 17.7.3; teleport_18: 18.1.1 -> 18.2.0

JuliusFreudenberger opened the pull request:
Changelogs: https://github.com/gravitational/teleport/releases/tag/v17.5.5 https://github.com/gravitational/teleport/releases/tag/v17.5.6 https://github.com/gravitational/teleport/releases/tag/v17.6.0 https://github.com/gravitational/teleport/releases/tag/v17.7.0 [ttps://github.com/gravitational/teleport/releases/tag/v17.7.1](https://github.com/gravitational/teleport/releases/tag/v17.7.1) https://github.com/gravitational/teleport/releases/tag/v17.7.2 https://github.com/gravitational/teleport/releases/tag/v17.7.3
Diff: https://github.com/gravitational/teleport/compare/v17.5.4...v17.7.3

Changelog: https://github.com/gravitational/teleport/releases/tag/v16.5.14 https://github.com/gravitational/teleport/releases/tag/v16.5.15
Diff: https://github.com/gravitational/teleport/compare/v16.5.13...v16.5.15

Changelogs: https://github.com/gravitational/teleport/releases/tag/v18.1.2 https://github.com/gravitational/teleport/releases/tag/v18.1.3 https://github.com/gravitational/teleport/releases/tag/v18.1.4 https://github.com/gravitational/teleport/releases/tag/v18.1.5 https://github.com/gravitational/teleport/releases/tag/v18.1.6 https://github.com/gravitational/teleport/releases/tag/v18.1.7 https://github.com/gravitational/teleport/releases/tag/v18.1.8 https://github.com/gravitational/teleport/releases/tag/v18.2.0
Diff: https://github.com/gravitational/teleport/compare/v18.1.1...v18.2.0


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

JuliusFreudenberger committed:
teleport_16: 16.5.13 -> 16.5.14

Changelog: https://github.com/gravitational/teleport/releases/tag/v16.5.14
Diff: https://github.com/gravitational/teleport/compare/v16.5.13...v16.5.14

JuliusFreudenberger committed:
teleport_17: 17.5.4 -> 17.7.0

Changelogs: https://github.com/gravitational/teleport/releases/tag/v17.5.5 https://github.com/gravitational/teleport/releases/tag/v17.5.6 https://github.com/gravitational/teleport/releases/tag/v17.6.0 https://github.com/gravitational/teleport/releases/tag/v17.7.0
Diff: https://github.com/gravitational/teleport/compare/v17.5.4...v17.7.0

Also add fix for error: "ERROR: mkdir /homeless-shelter: permission denied"

justinas reviewed (approved):
17.7.3 is now available, but I'd rather this got merged as-is, as it's been open for a bit now.

JuliusFreudenberger commented:
I have that already locally but waited for 16 to also have a release with this change.
This now happened, but I am currently on vacation. On Monday latest I will push the changes. If you want to go for it earlier, please feel free.

justinas commented:
> I have that already locally but waited for 16 to also have a release with this change. This now happened, but I am currently on vacation. On Monday latest I will push the changes. If you want to go for it earlier, please feel free.

I'll leave it up to you if you want to append to this PR! Feel free to cherry-pick things from https://github.com/NixOS/nixpkgs/compare/master...justinas:nixpkgs:teleport-18-2-0

Hope you had a great vacation :slightly_smiling_face:

JuliusFreudenberger committed:
teleport: rename references to rdp library to `librdpclient.h`

The file was renamed upstream, so the patch and import have to be
updated respectively.

JuliusFreudenberger committed:
teleport_18: 18.1.1 -> 18.2.0

Changelogs: https://github.com/gravitational/teleport/releases/tag/v18.1.2 https://github.com/gravitational/teleport/releases/tag/v18.1.3 https://github.com/gravitational/teleport/releases/tag/v18.1.4 https://github.com/gravitational/teleport/releases/tag/v18.1.5 https://github.com/gravitational/teleport/releases/tag/v18.1.6 https://github.com/gravitational/teleport/releases/tag/v18.1.7 https://github.com/gravitational/teleport/releases/tag/v18.1.8 https://github.com/gravitational/teleport/releases/tag/v18.2.0
Diff: https://github.com/gravitational/teleport/compare/v18.1.1...v18.2.0

JuliusFreudenberger committed:
teleport_17: 17.7.0 -> 17.7.3

Changelogs: https://github.com/gravitational/teleport/releases/tag/v17.7.1 https://github.com/gravitational/teleport/releases/tag/v17.7.2 https://github.com/gravitational/teleport/releases/tag/v17.7.3
Diff: https://github.com/gravitational/teleport/compare/v17.7.0...v17.7.3

JuliusFreudenberger committed:
teleport_16: 16.5.14 -> 16.5.15

Changelog: https://github.com/gravitational/teleport/releases/tag/v16.5.15
Diff: https://github.com/gravitational/teleport/compare/v16.5.14...v16.5.15
